app/src/main/java/io/gnosis/safe/utils/StringUtils.kt

```diff
@@ -67,6 +73,25 @@ fun Solidity.Address.formatEthAddress(context: Context, prefixLength: Int = 6, s
     }
 }
 
+@StringRes
+fun Solidity.Address?.fallBackHandlerLabel(): Int =
+    if (this == DEFAULT_FALLBACK_HANDLER) {
+        R.string.default_fallback_handler
+    } else {
+        R.string.unknown_fallback_handler
+    }
+
+@StringRes
+fun Solidity.Address.implementationVersion(): Int {
```

Comment:
it would be cleaner to move this to SafeRepository. Then knowledge of the safe, contracts, and versions would be encapsulated in one place